core/src/main/java/io/grpc/internal/AbstractClientStream2.java

```diff
@@ -384,6 +384,7 @@ public void dispose() {
       headers = null;
     }
 
+    // This framer sends everything in the header. So the methods below do not apply. 
```

Comment:
We do call writeHeaders API only so I think it's related? Any suggestions on rephrasing it?